cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -280,11 +280,12 @@ func (g *gitHubRepository) getVersions() ([]string, error) {
 
 	// get all the releases
 	// NB. currently Github API does not support result ordering, so it not possible to limit results
-	var releases []*github.RepositoryRelease
+	var allReleases []*github.RepositoryRelease
 	var retryError error
 	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
 		var listReleasesErr error
-		releases, _, listReleasesErr = client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, nil)
+		// Get the first page of GitHub releases
+		releases, response, listReleasesErr := client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, nil)
```

Comment:
is there any unit test that can be added to test the bug this is fixing?

Reply:
I did look into the test file for this source code, and it seems like it's using a fake request handler. Would you like me to add a test case for this there?

Reply:
The source code itself has been validated locally [here](https://github.com/kubernetes-sigs/cluster-api/issues/8233#issuecomment-1456989355).

Reply:
I have modified the getVersions Test method to construct a fake releases listing and emulate three different scenarios, one where an valid page number is given, one where an invalid page number is given and one which fetches all the (fake) releases. This is essentially mocking the ListReleases API call.